### PR TITLE
Migrated backup.py to go

### PR DESCRIPTION
**Pull request by arindamnayak**
@deepthi , I have created PR against planetscale repo as there are some pending things to be done.
- Some assertions for db variables and status
- Individual test runs successfully(`git test -run <name>`),  but not when it runs all.

### CODE REVIEW

File: go/test/endtoend/backup/backup_test.go [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+
+func TestReplicaBackup(t *testing.T) {
+	testBackup(t, "replica")
+}
+
+func TestRdonlyBackup(t *testing.T) {
+	testBackup(t, "rdonly")
+}
+
+//test_backup will:
```
Comment (deepthi) [inline go/test/endtoend/backup/backup_test.go]: This comment is in the wrong place

File: go/test/endtoend/backup/backup_test.go [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+
+	verifyRowsInTablet(t, replica2, 2)
+	verifyLocalMetadata(t, "replica")
+	verifyAfterRemovingBackupNoBackupShouldBePresent(t, backups)
+
+	replica2.VttabletProcess.TearDown()
+	master.VttabletProcess.QueryTablet("DROP TABLE vt_insert_test", keyspaceName, true)
+}
+
+//    Test a master and slave from the same backup.
```
Comment (deepthi) [inline go/test/endtoend/backup/backup_test.go]: replace slave -> replica

File: go/test/endtoend/backup/backup_test.go [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+
+	_, err = os.Stat(path.Join(master.VttabletProcess.Directory, "restore_in_progress"))
+	assert.True(t, os.IsNotExist(err))
+
+	verifyRowsInTablet(t, master, 3)
+}
+
+//
+//Test backup flow.
+//
```
Comment (deepthi) [inline go/test/endtoend/backup/backup_test.go]: remove these 3 lines

**Review by deepthi — COMMENTED** [review]
Looks good so far except for 
- some changes to comments
- some github check failures that need to be fixed.

**Comment by arindamnayak** [comment]
Thanks @deepthi , I have implemented the suggestions, I have created PR against upstream, so closing this one.

**Pull request closed without merging**